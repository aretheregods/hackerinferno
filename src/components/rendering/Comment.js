export function Comment(props) {
    
    const obj = props.obj
    const more_comments = props.nested_comments

    return (
        <div>
            <div className="comment">
                <div className="comment-details">
                    <a href={`/user/${obj["user"]}`}>
                        <p>{obj["user"]}</p>
                    </a>
                    <p className="comment-detaiil-child">&nbsp;{obj["time_ago"]}</p>
                </div>
                <div className="the-comment" dangerouslySetInnerHTML={{__html:obj["content"]}}></div>
            </div>
            {more_comments(obj)}
        </div>
    )
}
