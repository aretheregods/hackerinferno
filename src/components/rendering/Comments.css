#comments-container {
	margin: 1vh 2vw 1vh 2vw;
}

#story-url {
	text-decoration: none;
	color: red;
}

#comments-hr {
	min-width: 20vw;
	max-width: 50vw;
	color: red;
}

.comments-list {
	list-style: none;
	max-width: 100%;
	margin: 1vh 2vw 1vh 2vw;
}

.comments-list li {
	max-width: 100%;
	overflow-wrap: break-word;
}

.comment-details {
	color: red;
	display: flex;
}

.comment-details a,
.comment-details a:active,
.comment-details a:hover,
.comment-details a:visited {
	text-decoration: none;
	color: red;
}

.comment-detail-child {
	margin-right: 10px;
}

.the-comment {
	overflow-wrap: break-word;
}

pre,
code {
	overflow-wrap: break-word;
	white-space: pre-wrap;
}