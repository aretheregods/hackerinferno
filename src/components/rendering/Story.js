export function Story(props) {

    const data = props.story


    return (
        <div id="story-info">
            <a id="story-url" href={data.url}>{data.title}</a>
            <div>
                <p>{data.user && `by ${data.user} ${data.time_ago} | ${data.points || "0"} ${String.fromCharCode(9734)}`}</p>
            </div>
        </div>
    )
}
