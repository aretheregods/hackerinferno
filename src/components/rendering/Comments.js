import { icv, he, DC } from '../constants/Constants';
import { hn_api_pages } from '../constants/API';
import './Comments.css';


function Hacker_Comments(props) {

    const data = props.component_data;
    const comments_state = data.comments_count

	function make_comments_list(hacker_obj) {

		// Get the values of the data object
		const j = Object.values(hacker_obj).map(function find_comments(i, k) {

			// What will be returned inside each list
			var response = [];

			// Check each data object to see if it's an array with stuff inside
			// If it is, it's the one we want
			if(typeof i === 'object' && i !== null) {

				// Recursively check each object in the array in the same fashion
				// And make list items out of the data
				for(let obj of i) {
					response.push(icv(he, 'div', null, [
						icv(he, 'div', 'comment', [
							icv(he, 'div', 'comment-details', [
								icv(he, 'a', 'comment-detail-child', icv(he, 'p', null, `${obj["user"]}`), {href: `/user/${obj["user"]}`}),
								icv(he, 'p', 'comment-detail-child',`${obj["time_ago"]}`)
							]),
							icv(he, 'div', 'the-comment', null, {dangerouslySetInnerHTML: {__html: obj["content"]}})
						]),
						make_comments_list(obj)])
					)
				}
			}

			return icv(he, 'div', 'comments-list', response, {style: `margin-left: ${k + 1}`})

		})

		return j;

	};

    const story_data = icv(he, 'div', null, [
            icv(he, 'div', null, [
            	icv(he, 'a', null, data.title, {href: data.url, id: "story-url"}),
            	icv(he, 'div', null, [
                    icv(he, 'p', null, [
                    	data.user && "by " + data.user + " ",
                    	data.time_ago + " | ",
                    	`${data.points || "0"} ${String.fromCharCode(9734)}`
                    ])
            	])
            ], {id: 'story-info'}),
            icv(he, 'div', null, [
            	    icv(he, 'div', null, [
            	    	icv(he, 'hr', null, null, {id: 'comments-hr'}),
            	    	icv(he, 'h4', null, `${data.comments ? comments_state : "Loading..."} Comments`)
            	    ]),
            	    make_comments_list(data)
            	], {id: 'comments-list'}
            )
        ], {id: 'comments-container'}
    )

    return story_data;

}

export const Comments_Hacker = DC(Hacker_Comments)(hn_api_pages)('item');
