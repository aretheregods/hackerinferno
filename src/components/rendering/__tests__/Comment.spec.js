import { render } from 'inferno';
import { renderToSnapshot } from 'inferno-test-utils';
import { Comment } from '../Comment';
import { make_comments_list } from '../Comments.js';
import { Story } from '../Story';
import { HackerComments } from '../Comments';

const comment = {
    id: 1,
    user: "me",
    time_ago: "2 hours ago",
    points: 5,
    comments_count: 3,
    comments: [
        {
            user: "a",
            time_ago: "1 hour ago",
            content: "<p>A</p>",
            comments_count: 2,
            comments: [
                {
                    id: 2,
                    user: "b",
                    time_ago: "45 minutes ago",
                    content: "<p>B</p>",
                    comments_count: 0,
                    comments: []
                },
                {
                    id: 3,
                    user: "c",
                    time_ago: "30 minutes ago",
                    content: "<p>C</p>",
                    comments_count: 0,
                    comments: []
                }
            ]
        }
    ]
}

it('should render the top level comment and indent the nested comments', () => {
    const snapshot = renderToSnapshot(<Comment obj={comment} nested_comments={make_comments_list}/>);
    expect(snapshot).toMatchSnapshot();
})

it('should render the Story info and the list of associated comments', () => {
    const snapshot = renderToSnapshot(<HackerComments component_data={comment}/>);
    expect(snapshot).toMatchSnapshot();
})

it('should return an array of comments with a subarray of subcomments', () => {
    const snapshot = renderToSnapshot(<HackerComments component_data={comment}/>);
    expect(snapshot).toMatchSnapshot()
})

