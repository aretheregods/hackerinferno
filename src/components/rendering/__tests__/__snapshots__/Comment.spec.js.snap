// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Story info and the list of associated comments 1`] = `
<div
  id="comments-container"
>
  <div
    id="story-info"
  >
    <a
      id="story-url"
    />
    <div>
      <p>
        by me 2 hours ago | 5 â˜†
      </p>
    </div>
  </div>
  <div
    id="comments-list"
  >
    <div>
      <hr
        id="comments-hr"
      />
      <h4>
         Comments
      </h4>
    </div>
    <div
      className="comments-list"
      style={
        Object {
          "margin-left": "7px",
        }
      }
    >
      <div>
        <div
          className="comment"
        >
          <div
            className="comment-details"
          >
            <a
              href="/user/a"
            >
              <p>
                a
              </p>
            </a>
            <p
              className="comment-detaiil-child"
            >
               
              1 hour ago
            </p>
          </div>
          <div
            className="the-comment"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p>A</p>",
              }
            }
          />
        </div>
        <div
          className="comments-list"
          style={
            Object {
              "margin-left": "6px",
            }
          }
        >
          <div>
            <div
              className="comment"
            >
              <div
                className="comment-details"
              >
                <a
                  href="/user/b"
                >
                  <p>
                    b
                  </p>
                </a>
                <p
                  className="comment-detaiil-child"
                >
                   
                  45 minutes ago
                </p>
              </div>
              <div
                className="the-comment"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<p>B</p>",
                  }
                }
              />
            </div>
            <div
              className="comments-list"
              style={
                Object {
                  "margin-left": "7px",
                }
              }
            />
          </div>
          <div>
            <div
              className="comment"
            >
              <div
                className="comment-details"
              >
                <a
                  href="/user/c"
                >
                  <p>
                    c
                  </p>
                </a>
                <p
                  className="comment-detaiil-child"
                >
                   
                  30 minutes ago
                </p>
              </div>
              <div
                className="the-comment"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<p>C</p>",
                  }
                }
              />
            </div>
            <div
              className="comments-list"
              style={
                Object {
                  "margin-left": "7px",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render the top level comment and indent the nested comments 1`] = `
<div>
  <div
    className="comment"
  >
    <div
      className="comment-details"
    >
      <a
        href="/user/me"
      >
        <p>
          me
        </p>
      </a>
      <p
        className="comment-detaiil-child"
      >
         
        2 hours ago
      </p>
    </div>
    <div
      className="the-comment"
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
    />
  </div>
  <div
    className="comments-list"
    style={
      Object {
        "margin-left": "7px",
      }
    }
  >
    <div>
      <div
        className="comment"
      >
        <div
          className="comment-details"
        >
          <a
            href="/user/a"
          >
            <p>
              a
            </p>
          </a>
          <p
            className="comment-detaiil-child"
          >
             
            1 hour ago
          </p>
        </div>
        <div
          className="the-comment"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>A</p>",
            }
          }
        />
      </div>
      <div
        className="comments-list"
        style={
          Object {
            "margin-left": "6px",
          }
        }
      >
        <div>
          <div
            className="comment"
          >
            <div
              className="comment-details"
            >
              <a
                href="/user/b"
              >
                <p>
                  b
                </p>
              </a>
              <p
                className="comment-detaiil-child"
              >
                 
                45 minutes ago
              </p>
            </div>
            <div
              className="the-comment"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>B</p>",
                }
              }
            />
          </div>
          <div
            className="comments-list"
            style={
              Object {
                "margin-left": "7px",
              }
            }
          />
        </div>
        <div>
          <div
            className="comment"
          >
            <div
              className="comment-details"
            >
              <a
                href="/user/c"
              >
                <p>
                  c
                </p>
              </a>
              <p
                className="comment-detaiil-child"
              >
                 
                30 minutes ago
              </p>
            </div>
            <div
              className="the-comment"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>C</p>",
                }
              }
            />
          </div>
          <div
            className="comments-list"
            style={
              Object {
                "margin-left": "7px",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
